app/models/breakeven_table_question.rb

```diff
@@ -18,76 +25,202 @@ def initialize(breakeven)
-  #     { name: "Worker owners", amount: 28_000 },
-  #     { name: "Employees", amount: 10_000 },
-  #     { name: "Sales", amount: 10_000 },
-  #     { name: "Utilities", amount: 2_000 },
-  #     { name: "Insurance", amount: 1_000 },
+  #     { name: 'Rent', amount: 15_000.0 },
+  #     { name: 'Worker owners', amount: 28_000.0 },
+  #     { name: 'Employees', amount: 10_000.0 },
+  #     { name: 'Sales', amount: 10_000 },
+  #     { name: 'Utilities', amount: 2_000.0 },
+  #     { name: 'Insurance', amount: 1_000.0 },
   #   ],
   #   total_fixed_costs: 66_000,
+  #   total_fixed_costs_rampup: [66_000, 66_000, 66_000, 66_000],
   #   net_margin: 0,
+  #   net_margin_rampup: [-49_500, -33_000, -16_500, 0],
+  #   periods: 4,
+  #   units: 'Months'
   # }
   def report
     return if blank?
 
     report = {
-      revenue: [],
-      total_revenue: 0,
-      cogs: [],
-      total_cogs: 0,
-      gross_margin: 0,
-      fixed_costs: [],
-      total_fixed_costs: 0,
-      net_margin: 0,
+      revenue: revenue,
+      total_revenue: total_revenue,
+      cogs: cogs,
+      total_cogs: total_cogs,
+      gross_margin: gross_margin,
+      fixed_costs: fixed_costs,
+      total_fixed_costs: total_fixed_costs,
+      net_margin: net_margin,
+      periods: periods,
+      units: units,
     }
 
-    @breakeven['products'].each do |product|
-      %w(quantity price cost).each { |key| product[key] = product[key].to_f }
-      report[:revenue] << {
-        name: product['name'],
-        quantity: product['quantity'],
-        amount: product['price'],
-        total: product['price'] * product['quantity'],
-      }
-      report[:cogs] << {
-        name: product['name'],
-        quantity: product['quantity'],
-        amount: product['cost'],
-        total: product['cost'] * product['quantity'],
-      }
-    end
-    report[:total_revenue] = report[:revenue].map { |i| i[:total] }.sum
-    report[:total_cogs] = report[:cogs].map { |i| i[:total] }.sum
-    report[:gross_margin] = report[:total_revenue] - report[:total_cogs]
-    @breakeven['fixed_costs'].each do |cost|
-      report[:fixed_costs] << { name: cost['name'], amount: cost['amount'].to_f }
-    end
-    report[:total_fixed_costs] = report[:fixed_costs].map { |i| i[:amount] }.sum
-    report[:net_margin] = report[:gross_margin] - report[:total_fixed_costs]
+    rampup = {
+      total_revenue_rampup: total_revenue_rampup,
+      total_cogs_rampup: total_cogs_rampup,
+      gross_margin_rampup: gross_margin_rampup,
+      total_fixed_costs_rampup: total_fixed_costs_rampup,
+      net_margin_rampup: net_margin_rampup,
+    }
 
-    report
+    return report unless periods > 1
+    report.merge(rampup)
   end
 
   def blank?
     return true if @breakeven.blank?
-    @breakeven['products'].blank? && @breakeven['fixed_costs'].blank?
+    data_hash[:products].blank? && data_hash[:fixed_costs].blank?
+  end
+
+  def revenue
+    @revenue ||= data_hash[:products].map do |product|
```

Comment:
Memoized to prevent performance issues when this method is called multiple times.